util-app/src/main/scala/com/twitter/app/App.scala

```diff
@@ -118,7 +127,13 @@ trait App extends Closable with CloseAwaitably {
    *
    *       In all cases, the close deadline is enforced.
    */
-  final def closeOnExitLast(closable: Closable): Unit = lastExits.add(closable)
+  final def closeOnExitLast(closable: Closable): Unit = {
+    lastExits.add(closable)
+
+    if (closeDeadline != Time.Top) {
```

Comment:
it's possible for this to create a race condition where this is closed before the "first" closables have started.  maybe we should fix this with mutexes?